qiskit/assembler/assemble_schedules.py

```diff
@@ -246,20 +246,29 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
     Raises:
         QiskitError: If the instructions do not satisfy the measurement map.
     """
+    sorted_inst_map = sorted(instruction_map.items(), key=lambda item: item[0])
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
-        measured_qubits = set()
-        for inst in instrs:
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
-
+    all_measured_qubits = [inst.channel.index for _, insts in sorted_inst_map
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
+        inst_end_time = instr[0][0] + instr[0][1]
+        if sorted_inst_map[idx+1][0][0] < inst_end_time:
+            instr_qubits = {inst.channel.index for inst in instr[1]}
+            next_instr_qubits = {inst.channel.index for inst in sorted_inst_map[idx+1][1]}
+            for meas_set in meas_map_sets:
+                common_instr_qubits = instr_qubits.intersection(meas_set)
+                common_next = next_instr_qubits.intersection(meas_set)
+                if common_instr_qubits and common_next:
+                    raise QiskitError('Qubits {} and {} in the measurement map: {} was '
+                                      'acquired disjointly'.format(common_instr_qubits,
+                                                                   common_next, meas_map))
```

Comment:
```suggestion
                    raise QiskitError('Qubits {} and {} are in the same measurement grouping: {}. '
                                      'They must either be acquired at the same time, or disjointly. '
                                      'Instead, they were acquired at times: {}-{} and {}-{}'.format(common_instr_qubits,
                                                                   common_next, meas_map,
                                                                   instr[0][0], inst_end_time,
                                                                   next_inst_time, next_inst_time + next_inst[0][1]))
```
Going to recommend also adding a couple more variables to make it more readable and to make this suggestion ^ execute

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca